### PR TITLE
Add step-specific hashes to the URL in the OnboardingWizard

### PR DESCRIPTION
**Pull request by Dieterrr**
## Summary

This PR can be summarized in the following changelog entry:

* Add step-specific hashes to the URL in the OnboardingWizard to make navigation easier.

## Relevant technical choices:

* N/A

## Test instructions

This PR can be tested by following these steps:

* See https://github.com/Yoast/wordpress-seo/pull/12162

## UI changes
* [ ] This PR changes the UI in the plugin. I have added the 'UI change' label to this PR.

Fixes #[12158](https://github.com/Yoast/wordpress-seo/issues/12158)

### CODE REVIEW

File: composites/OnboardingWizard/OnboardingWizard.js [hunk 1]
```diff
@@ -259,6 +276,42 @@ class OnboardingWizard extends React.Component {
...
+	 *
+	 * If we wouldn't check against the current hash, it would lead to double hashes when using the browser's previous
+	 * and next buttons.
+	 *
+	 * @param {Object} prevProps The previous props.
+	 * @param {Object} prevState The previous state.
+	 *
+	 * @returns {void}
+	 */
+	getSnapshotBeforeUpdate( prevProps, prevState ) {
```
Comment (andizer) [inline composites/OnboardingWizard/OnboardingWizard.js]: What is the usecase for this function, it isn't used?

File: composites/OnboardingWizard/OnboardingWizard.js [hunk 2]
```diff
@@ -131,6 +134,11 @@ class OnboardingWizard extends React.Component {
 	 * @returns {Object}  The first step object
 	 */
 	getFirstStep( steps ) {
+		const firstStep = this.props.startingStep;
+		if ( firstStep !== "" ) {
+			return firstStep;
```
Comment (andizer) [inline composites/OnboardingWizard/OnboardingWizard.js]: This line isn't covered?

File: composites/OnboardingWizard/OnboardingWizard.js [hunk 3]
```diff
@@ -204,7 +212,16 @@ class OnboardingWizard extends React.Component {
 	 * @returns {Object} The current step.
 	 */
 	getCurrentStep() {
-		return this.state.steps[ this.state.currentStepId ];
+		const steps = this.state.steps;
+		const currentStep = steps[ this.state.currentStepId ];
+
+		// If the currentStep is undefined because the stepId is invalid, return the first step of the Wizard.
+		if ( isUndefined( currentStep ) ) {
+			const firstStepId = Object.keys( steps )[ 0 ];
```
Comment (andizer) [inline composites/OnboardingWizard/OnboardingWizard.js]: These lines aren't covered?

File: composites/OnboardingWizard/OnboardingWizard.js (hunk 1 above)
Comment (IreneStr) [reply composites/OnboardingWizard/OnboardingWizard.js]: This runs before componentDidMount, and is one of the React lifecycle methods.

File: composites/OnboardingWizard/OnboardingWizard.js (hunk 3 above)
Comment (IreneStr) [reply composites/OnboardingWizard/OnboardingWizard.js]: I've created a technical debt issue: https://github.com/Yoast/yoast-components/issues/824

File: composites/OnboardingWizard/OnboardingWizard.js (hunk 2 above)
Comment (IreneStr) [reply composites/OnboardingWizard/OnboardingWizard.js]: I've created a technical debt issue: https://github.com/Yoast/yoast-components/issues/824

**Comment by Dieterrr** [comment]
Acceptance done 👍

**Pull request merged**